araproc/analysis/standard_reco.py

```diff
@@ -330,3 +384,187 @@ def lookup_arrival_time(self,
                                                                                        phi_bin.value
                                                                                        )
         return arrival_time
+
+    def get_surface_corr_max(self, corr_map, z_thresh=-10):
+        """
+        Calculates the maximum surface correlation value within a specified theta range.
+        The theta range is determined based on the average antenna z-coordinate and a threshold.
+
+        Parameters
+        ----------
+        corr_map : dict
+            Dictionary containing:
+            - 'map': ROOT.TH2D histogram with correlation values mapped by theta and phi.
+            - 'corr': Overall peak correlation value.
+            - 'radius': Correlation radius for the station (in meters).
+        z_thresh : float, optional
+            Depth threshold for defining the theta range lower bound. Default is -10 m.
+
+        Returns
+        -------
+        max_surf_corr : float
+            Maximum surface correlation within the theta range.
+        max_theta : float
+            Theta angle corresponding to max surface correlation.
+        max_phi : float
+            Phi angle corresponding to max surface correlation.
+        """
+        # Extract the radius and validate
+        radius = corr_map.get("radius", None)
+        if radius is None:
+            raise ValueError("Radius not found in corr_map.")
+        radius = float(radius)
+
+        # Get antenna z-coordinates and calculate average
+        geom_tool = ROOT.AraGeomTool.Instance()
+        z_coords = [geom_tool.getStationInfo(self.station_id).getAntennaInfo(ant).antLocation[2] 
+                    for ant in range(self.num_channels)]
+        avg_z = sum(z_coords) / len(z_coords)
+
+        # Check if surface is visible at the given radius
+        if radius < (abs(avg_z) + z_thresh):
+            raise RuntimeError("Surface not visible: radius is smaller than average antenna z-coordinate.")
+
+        # Define theta range
+        theta_surface = math.degrees(math.asin(abs(avg_z) / radius))
```

Comment:
Thinking out loud, should we return degrees or radians? I guess the correlation maps are in degrees?

Reply:
Degrees. Yes, the corr. maps are in degrees.

Reply:
Sounds good!